### ISSUE TITLE
`clip-path` aliasing

### ISSUE THREAD

**Issue by stanio** [weisJ/jsvg#59]
`clip-path-test.svg`

```xml
<svg xmlns="http://www.w3.org/2000/svg" width="100" height="100" viewBox="0 0 200 200">
  <path d="m90 25v65h-65v20h65v65h20v-65h65v-20h-65v-65h-20z"
      fill="lightgreen" stroke="indigo" stroke-linejoin="round" stroke-width="30"
      paint-order="stroke fill" clip-path="url(#aperture)" />
  <clipPath id="aperture">
    <path d="m0 0v200h200v-200h-200zm100 70a30 30 0 0 1 30 30 30 30 0 0 1-30 30 30 30 0 0 1-30-30 30 30 0 0 1 30-30z" />
  </clipPath>
  <circle cx="100" cy="100" r="5" fill="red" />
</svg>
```

```java
import java.io.File;

import java.awt.Color;
import java.awt.Dimension;
import java.awt.Graphics2D;
import java.awt.RenderingHints;
import java.awt.image.BufferedImage;

import javax.imageio.ImageIO;

import com.github.weisj.jsvg.SVGDocument;
import com.github.weisj.jsvg.attributes.ViewBox;
import com.github.weisj.jsvg.parser.SVGLoader;

public class ClipPathTest {

    public static void main(String[] args) throws Exception {
        SVGDocument svg = new SVGLoader()
                .load(ClipPathTest.class.getResource("clip-path-test.svg"));
        Dimension size = new Dimension(100, 100);
        BufferedImage image = new BufferedImage(size.width, size.height,
                                                BufferedImage.TYPE_INT_RGB);
        Graphics2D g = image.createGraphics();
        g.setRenderingHint(RenderingHints.KEY_ANTIALIASING,
                           RenderingHints.VALUE_ANTIALIAS_ON);
        g.setBackground(Color.WHITE);
        g.clearRect(0, 0, size.width, size.height);
        svg.render(null, g, new ViewBox(size.width, size.height));
        g.dispose();
        ImageIO.write(image, "png", new File("clip-path-test.png"));
        System.out.println("Done.");
    }

}
```

Actual | Expected
--- | ---
![aliased](https://github.com/weisJ/jsvg/assets/1247730/b14fb67b-6291-48e7-b92f-1bef7a7ec612) | ![smooth](https://github.com/weisJ/jsvg/assets/1247730/612fe27f-e6ec-4bc3-94da-2821e9c64b8e)

---

I've found #6 closed as completed in 6ecafcfaf63db84901514928a5124a34f166addc but it likely solves a different scenario.

**Comment by weisJ** [comment]
I have implemented the relevant soft clipping in #60. To enable it you have to set `SVGRenderingHints.KEY_SOFT_CLIPPING` to `SVGRenderingHints.VALUE_SOFT_CLIPPING_ON` in the graphics rendering hints. In the future this will be enabled by default, but I first want to implement some more fast-paths and make sure it works in all scenarios.

You can test it out using the latest snapshot release.

**Comment by stanio** [comment]
Tested with current [1.4.0-SNAPSHOT](https://oss.sonatype.org/content/repositories/snapshots/com/github/weisj/jsvg/1.4.0-SNAPSHOT/) (jsvg-1.4.0-20240213.002554-80):

```gradle
repositories {
    maven {
        url 'https://oss.sonatype.org/content/repositories/snapshots/'
    }
}

dependencies {
    implementation 'com.github.weisj:jsvg:latest.integration'
}
```

```java
        g.setRenderingHint(SVGRenderingHints.KEY_SOFT_CLIPPING,
                           SVGRenderingHints.VALUE_SOFT_CLIPPING_ON);
```

Works as advertised – thank you!

**Comment by weisJ** [comment]
Now available in version `1.4.0`
